### ISSUE TITLE
Bug report: `pip` install of the repo does not work on Windows

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [materialsproject/pymatgen-io-validation#88]
### Problem

`pip` install of fa3bb8bd274c1bb4caab8a269e082aa5f57e40aa does not work on Windows.

```
 × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      Traceback (most recent call last):
        File "C:\Users\asros\miniconda\envs\quacc\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\asros\miniconda\envs\quacc\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\asros\miniconda\envs\quacc\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\asros\AppData\Local\Temp\pip-build-env-zshwb9fi\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\asros\AppData\Local\Temp\pip-build-env-zshwb9fi\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "C:\Users\asros\AppData\Local\Temp\pip-build-env-zshwb9fi\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 11, in <module>
        File "C:\Users\asros\miniconda\envs\quacc\Lib\encodings\cp1252.py", line 23, in decode
          return codecs.charmap_decode(input,self.errors,decoding_table)[0]
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 2056: character maps to <undefined>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```

### Proposed Solution

Fix line-ending issues with `ruff` (?). Also add a Windows CI runner.

**Comment by esoteric-ephemera** [comment]
Thanks @Andrew-S-Rosen! Should be fixed in [e84dcc0](https://github.com/materialsproject/pymatgen-io-validation/commit/e84dcc06b0959f88de77cf3a9c9ff1fa78614560), can you check that it's working for you?

**Comment by Andrew-S-Rosen** [comment]
Mission accomplished. Build successful. 🚀
